pre-commit hook does not work

https://csharpier.com/docs/Pre-commit says you can use csharpier with pre-commit, but this functionality doesn't work - after including the csharpier hook as per the instructions, the hook fails to install:

```
$ pre-commit run
[INFO] Installing environment for https://github.com/belav/csharpier.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('C:\\Program Files\\dotnet\\dotnet.EXE', 'tool', 'install', '--configfile', 'C:\\Users\\Lucas\\AppData\\Local\\Temp\\tmpcrqgampz\\nuget.config', '--tool-path', 'C:\\Users\\Lucas\\.cache\\pre-commit\\reponbup7ntf\\dotnetenv-default\\bin', '--add-source', 'pre-commit-build', 'CSharpier.Benchmarks')
return code: 1
stdout:
    C:\Users\Lucas\AppData\Local\Temp\a04c5a0c-40b7-4cba-8cc4-c616ccf96459\restore.csproj : error NU1101: Unable to find package BenchmarkDotNet. No packages exist with this id in source(s): C:\Users\Lucas\.cache\pre-commit\reponbup7ntf\pre-commit-build
    C:\Users\Lucas\AppData\Local\Temp\a04c5a0c-40b7-4cba-8cc4-c616ccf96459\restore.csproj : error NU1101: Unable to find package BenchmarkDotNet.Annotations. No packages exist with this id in source(s): C:\Users\Lucas\.cache\pre-commit\reponbup7ntf\pre-commit-build
    C:\Users\Lucas\AppData\Local\Temp\a04c5a0c-40b7-4cba-8cc4-c616ccf96459\restore.csproj : error NU1101: Unable to find package Microsoft.CodeAnalysis.CSharp. No packages exist with this id in source(s): C:\Users\Lucas\.cache\pre-commit\reponbup7ntf\pre-commit-build
    C:\Users\Lucas\AppData\Local\Temp\a04c5a0c-40b7-4cba-8cc4-c616ccf96459\restore.csproj : error NU1101: Unable to find package System.IO.Abstractions. No packages exist with this id in source(s): C:\Users\Lucas\.cache\pre-commit\reponbup7ntf\pre-commit-build
    C:\Users\Lucas\AppData\Local\Temp\a04c5a0c-40b7-4cba-8cc4-c616ccf96459\restore.csproj : error NU1101: Unable to find package System.Text.Json. No packages exist with this id in source(s): C:\Users\Lucas\.cache\pre-commit\reponbup7ntf\pre-commit-build

stderr:
    The tool package could not be restored.
    Tool 'csharpier.benchmarks' failed to install. This failure may have been caused by:

    * You are attempting to install a preview release and did not use the --version option to specify the version.
    * A package by this name was found, but it was not a .NET tool.
    * The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
    * You mistyped the name of the tool.

    For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool

Check the log at C:\Users\Lucas\.cache\pre-commit\pre-commit.log
```

--- a/.pre-commit-hooks.yaml
+++ b/.pre-commit-hooks.yaml
@@ -1,7 +1,0 @@
-- id: csharpier
-  name: Run CSharpier on C# files
-  entry: dotnet tool run dotnet-csharpier
-  language: dotnet
-  types:
-    - c#
-  description: CSharpier is an opinionated C# formatter inspired by Prettier.

--- a/Docs/Pre-commit.md
+++ b/Docs/Pre-commit.md
@@ -7,18 +7,60 @@ CSharpier can be used with a pre-commit hook to ensure that all staged files are
 
 ## [pre-commit](https://pre-commit.com)
 
+[Install pre-commit via your preferred Python package manager.](https://pre-commit.com/#install)
+[Run `pre-commit install` to install the Git hook scripts.](https://pre-commit.com/#3-install-the-git-hook-scripts)
+
+
+### [MegaLinter](https://megalinter.io/)
+
+CSharpier runs as part of
+[MegaLinter's pre-commit hooks](https://megalinter.io/latest/mega-linter-runner/#pre-commit-hook).
+
+
+### Standalone
+
+If you prefer not to run the other linters included in MegaLinter, you can
+alternatively run CSharpier as a local pre-commit hook.
+
 Add the following to your `.pre-commit-config.yaml`:
 
 ```yaml
 repos:
-    - repo: https://github.com/belav/csharpier
-      rev: 0.23.0 # Replace with the latest Git tag.
-      hooks:
-        - id: csharpier
+  - repo: local
+    hooks:
+      - id: dotnet-tool-restore
+        name: Install .NET tools
+        entry: dotnet tool restore
+        language: system
+        always_run: true
+        pass_filenames: false
+        stages:
+          - commit
+          - push
+          - post-checkout
+          - post-rewrite
+        description: Install the .NET tools listed at .config/dotnet-tools.json.
+      - id: csharpier
+        name: Run CSharpier on C# files
+        entry: dotnet tool run dotnet-csharpier
+        language: system
+        types:
+          - c#
+        description: CSharpier is an opinionated C# formatter inspired by Prettier.
 ```
 
-[Install pre-commit via your preferred Python package manager.](https://pre-commit.com/#install)
-[Run `pre-commit install` to install the Git hook scripts.](https://pre-commit.com/#3-install-the-git-hook-scripts)
+Add the following to your `.config/dotnet-tools.json`:
+
+```json
+{
+  "tools": {
+    "CSharpier": {
+      "version": "0.22.1",
+      "commands": ["dotnet-csharpier"]
+    }
+  }
+}
+```
 
 ## [Husky.Net](https://github.com/alirezanet/husky.net)
 

--- a/Src/Website/docs/Pre-commit.md
+++ b/Src/Website/docs/Pre-commit.md
@@ -7,18 +7,60 @@ CSharpier can be used with a pre-commit hook to ensure that all staged files are
 
 ## [pre-commit](https://pre-commit.com)
 
+[Install pre-commit via your preferred Python package manager.](https://pre-commit.com/#install)
+[Run `pre-commit install` to install the Git hook scripts.](https://pre-commit.com/#3-install-the-git-hook-scripts)
+
+
+### [MegaLinter](https://megalinter.io/)
+
+CSharpier runs as part of
+[MegaLinter's pre-commit hooks](https://megalinter.io/latest/mega-linter-runner/#pre-commit-hook).
+
+
+### Standalone
+
+If you prefer not to run the other linters included in MegaLinter, you can
+alternatively run CSharpier as a local pre-commit hook.
+
 Add the following to your `.pre-commit-config.yaml`:
 
 ```yaml
 repos:
-    - repo: https://github.com/belav/csharpier
-      rev: 0.23.0 # Replace with the latest Git tag.
-      hooks:
-        - id: csharpier
+  - repo: local
+    hooks:
+      - id: dotnet-tool-restore
+        name: Install .NET tools
+        entry: dotnet tool restore
+        language: system
+        always_run: true
+        pass_filenames: false
+        stages:
+          - commit
+          - push
+          - post-checkout
+          - post-rewrite
+        description: Install the .NET tools listed at .config/dotnet-tools.json.
+      - id: csharpier
+        name: Run CSharpier on C# files
+        entry: dotnet tool run dotnet-csharpier
+        language: system
+        types:
+          - c#
+        description: CSharpier is an opinionated C# formatter inspired by Prettier.
 ```
 
-[Install pre-commit via your preferred Python package manager.](https://pre-commit.com/#install)
-[Run `pre-commit install` to install the Git hook scripts.](https://pre-commit.com/#3-install-the-git-hook-scripts)
+Add the following to your `.config/dotnet-tools.json`:
+
+```json
+{
+  "tools": {
+    "CSharpier": {
+      "version": "0.22.1",
+      "commands": ["dotnet-csharpier"]
+    }
+  }
+}
+```
 
 ## [Husky.Net](https://github.com/alirezanet/husky.net)
 

